src/collection.js

```diff
@@ -365,10 +377,32 @@ export default class Collection {
         "max_target.data.last_modified": String(at),
       },
     });
+    return changes;
+  }
 
+  /**
+   * @private
+   */
+  async getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    // If history doesn't have enough data, reject with an error
+    const oldestHistory = await this.findHistoryOldestTimestamp();
+    if (!oldestHistory) {
+      throw new Error("History is not enabled.");
```

Comment:
It was part of a future plan indeed, I just want to have a proper logic implemented first.